### PR TITLE
[cmake] Enable IN_LIST compare policy

### PR DESCRIPTION
**Pull request by modocache**
Enable CMake policy CMP0057, which allows `if()` statements to use the `IN_LIST` operator. In addition, simplify several `if()` statements that used the `list(FIND ...)` operation instead.

This pull request was split out of https://github.com/apple/swift/pull/4972, which addresses [SR-1362](https://bugs.swift.org/browse/SR-1362).

### CODE REVIEW

**Comment by llvm-beanz** [comment]
Given that LLVM requires CMake 3.4.3 which has this by default wouldn't it be better to just raise Swift's CMake minimum version to 3.4.3 instead of setting the policy?

**Comment by gottesmm** [comment]
@llvm-beanz We do require cmake 3.4.3. Just no one has gotten around to changing the actual requirement in source b/c of potential policy differences.

Maybe that is the right fix here?

**Comment by modocache** [comment]
I'd love to do so, but perhaps in a separate commit? I could remove the `cmake_policy(SET CMP0057 NEW)` and bump the minimum version, leaving the `IN_LIST` changes alone. I'd prefer a future commit because bumping the minimum version leads to errors that I have yet to figure out.

**Comment by llvm-beanz** [comment]
@modocache, sounds reasonable to me.

**Comment by gottesmm** [comment]
@llvm-beanz @modocache Also SGTM

**Comment by gottesmm** [comment]
With that being said:

@swift-ci Please test

**Comment by gottesmm** [comment]
Hmmm... I actually wanted to do a clean cmake build.

**Comment by gottesmm** [comment]
@swift-ci Please clean test

**Comment by gottesmm** [comment]
Interesting. Libdispatch failed probably due to dependency race issues.

**Comment by gottesmm** [comment]
@swift-ci Please clean test os x platform

**Comment by modocache** [comment]
@swift-ci Please clean test os x platform

**Comment by modocache** [comment]
@swift-ci please test os x platform

**Comment by modocache** [comment]
@swift-ci please test os x platform

**Comment by modocache** [comment]
@swift-ci please test

**Comment by modocache** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
Hmm, I'm pretty baffled here. Running the exact same preset on my macOS machine succeeds. I don't understand where these Dispatch SDK overlay errors are coming from. Does anyone have any tips?

**Comment by gottesmm** [comment]
@jrose-apple This sounds like something up your alley.

**Comment by modocache** [comment]
For what it's worth, I have Xcode 8.0 (8A218a) installed, on macOS Sierra 10.12.

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -618,8 +618,7 @@ function(_add_swift_library_single target name)
   else()
     string(REPLACE swift "" module_name "${name}")
   endif()
-  list(FIND SWIFT_API_NOTES_INPUTS "${module_name}" overlay_index)
-  if(NOT ${overlay_index} EQUAL -1)
+  if("${modue_name}" IN_LIST SWIFT_API_NOTES_INPUTS)
```
Comment (jrose-apple) [inline cmake/modules/AddSwift.cmake]: Typo: missing "l". :-)

Comment (modocache) [inline cmake/modules/AddSwift.cmake]: And sure enough, a completely clean build did reproduce this on my macOS machine. Geez! 😬

**Comment by modocache** [comment]
@swift-ci please test

**Comment by modocache** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
OK, this finally seems good to merge. Look OK to you, @gottesmm?

**Comment by modocache** [comment]
Alright, I'm too excited for #4972 to let this wait for very long, so I'm going to go ahead and merge this. Sorry in advance if some of the buildbots fail due to incorrect CMake caches... 😇 

/cc @tfiala 🙇

**Comment by gottesmm** [comment]
No worries. LGTM.

**Comment by gottesmm** [comment]
I actually reviewed it on the bus, but the bus reached the destination. Sorry!

**Pull request merged**